File: frontend/js/system.js
// ==================== УПРАВЛЕНИЕ СИСТЕМОЙ ====================
App.systemInitialized = false;

App.initSystem = function() {
  if (App.systemInitialized) return;
  App.systemInitialized = true;

  // Обработчик кнопок тем и сброса
  document.addEventListener('click', async (e) => {
    const btn = e.target.closest('.system-btn');
    if (!btn) return;
    const action = btn.dataset.action;

    if (action === 'set-theme') {
      const mode = btn.dataset.mode;
      const res = await fetch('/set_theme', {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ mode })
      });
      const data = await res.json();
      showMessage('theme-message', data.message, data.success ? 'success' : 'error');
    }
    else if (action === 'reset-theme') {
      const res = await fetch('/reset_theme', { method: 'POST' });
      const data = await res.json();
      showMessage('theme-message', data.message, data.success ? 'success' : 'error');
    }
    else if (action === 'launch') {
      // старые кнопки быстрого запуска, если остались
      const appId = btn.dataset.app;
      const res = await fetch('/launch_app', {
        method: 'POST',
        headers: { 'Content-Type': 'application/json' },
        body: JSON.stringify({ app_id: appId })
      });
      const data = await res.json();
      showMessage('preset-message', data.message, data.success ? 'success' : 'error');
    }
  });

  // Обработчик запуска пресета
  document.getElementById('run-preset-btn')?.addEventListener('click', async () => {
    const select = document.getElementById('preset-select');
    const presetName = select.value;
    if (!presetName) return;
    const res = await fetch('/run_preset', {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify({ preset: presetName })
    });
    const data = await res.json();
    showMessage('preset-message', data.message, data.success ? 'success' : 'error');
  });

  // Загрузка списка пресетов
  async function loadPresets() {
    try {
      const res = await fetch('/get_presets');
      const data = await res.json();
      const select = document.getElementById('preset-select');
      if (!select) return;
      // Очищаем старые опции, кроме первой пустой
      select.innerHTML = '<option value="">-- Выберите пресет --</option>';
      for (const preset of data.presets) {
        const opt = document.createElement('option');
        opt.value = preset.name;
        opt.textContent = preset.name;
        select.appendChild(opt);
      }
    } catch (e) {
      console.error('Failed to load presets', e);
    }
  }

  function showMessage(elementId, text, type) {
    const el = document.getElementById(elementId);
    if (!el) return;
    el.textContent = text;
    el.style.color = type === 'error' ? '#e74c3c' : 'var(--text-secondary)';
    setTimeout(() => { el.textContent = ''; }, 4000);
  }

  loadPresets(); // загружаем сразу при инициализации
};

App.tabInitializers.system = App.initSystem;

File: backend/routes/system.py
from flask import Blueprint, request, jsonify
from backend.services.system_service import (
    set_windows_theme,
    reset_theme_to_default,
    launch_application,
    load_presets,
    run_preset
)

system_bp = Blueprint('system', __name__)

@system_bp.route('/set_theme', methods=['POST'])
def set_theme():
    mode = request.json.get('mode', 'dark')
    success, message = set_windows_theme(mode)
    return jsonify({'success': success, 'message': message})

@system_bp.route('/reset_theme', methods=['POST'])
def reset_theme():
    success, message = reset_theme_to_default()
    return jsonify({'success': success, 'message': message})

@system_bp.route('/launch_app', methods=['POST'])
def launch_app():
    app_id = request.json.get('app_id')
    success, message = launch_application(app_id)
    if not success:
        return jsonify({'success': False, 'message': message}), 400
    return jsonify({'success': True, 'message': message})

@system_bp.route('/get_presets', methods=['GET'])
def get_presets():
    presets = load_presets()
    return jsonify({'presets': [{'name': p['name']} for p in presets]})

@system_bp.route('/run_preset', methods=['POST'])
def run_preset_route():
    preset_name = request.json.get('preset')
    if not preset_name:
        return jsonify({'success': False, 'message': 'Не указан пресет'}), 400
    success, message = run_preset(preset_name)
    return jsonify({'success': success, 'message': message})